esp8266/machine_rtc.c

```diff
@@ -55,23 +56,96 @@ uint64_t pyb_rtc_alarm0_expiry; // in microseconds
 
 // RTC overflow checking
 STATIC uint32_t rtc_last_ticks;
+STATIC uint32_t rtc_last_cal;
+// Clock overflow checking
+STATIC uint32_t clk_last_ticks;
+STATIC uint64_t clk_offset;
+
+void esp_clk_set_us_since_2000(uint64_t nowus) {
+    // Get the current clock tick
+    clk_last_ticks = system_get_time();
+    // Set current time as base for future calculations
+    clk_offset = nowus;
+};
+
+uint64_t esp_clk_get_us_since_2000() {
+    uint64_t offset;
+    uint32_t clk_ticks;
+
+    offset = clk_offset;
+    clk_ticks = system_get_time();
+
+    if (clk_ticks >= clk_last_ticks) {
+      offset+= clk_ticks-clk_last_ticks;
+    } else {
+      // If overflow happened, assume 1 wrap-around and persist info for the new cycle
+      offset+= clk_ticks+~clk_last_ticks+1;
+      clk_last_ticks = clk_ticks;
+      clk_offset = offset;
+    }
+    return offset; 
+};
+
+void pyb_rtc_set_us_since_2000(uint64_t nowus) {
```

Comment:
I have move the pyb_rtc... functions here because I would like to use the set function in mp_hal_rtc_init(), and I think having the definition before use make make code reading a little easier. :)
The new pyb_rtc_... function are completely rewritten, so they have little relation to the old code. Their code logic are more like the esp_clk_... functions.